syntax = "proto2";
package Trd_ModifyOrder;

import "Common.proto";
import "Trd_Common.proto";

message C2S
{
	required Common.PacketID packetID = 1; //交易写操作防重放攻击
	required Trd_Common.TrdHeader header = 2; //交易公共参数头
	required uint64 orderID = 3; //订单号
	required int32 modifyOrderOp = 4; //修改订单的操作类型，参见Trd_Common.ModifyOrderOp的枚举定义
	optional bool forAll = 5; //操作是否对此业务账户全部订单操作，true是，false否(单个订单)
	
	//下面的字段仅在modifyOrderOp为ModifyOrderOp_Normal有效
	optional double qty = 8; //数量，港股2位小数精度
	optional double price = 9; //价格，港股3位小数精度
	//调整价格目前仅对港股有效，因为港股有价位，美股不需要
	optional bool adjustPrice = 10; //调整价格，如果价格不合法，调整到合法价位，true调整，false不调整
	optional double adjustSideAndLimit = 11; //调整方向和调整幅度百分比限制，正数代表向上调整，负数代表向下调整，取值代表幅度限制，如：0.015代表向上调整且幅度不超过1.5%；-0.01代表向下调整且幅度不超过1%
}

message S2C
{
	required Trd_Common.TrdHeader header = 1; //交易公共参数头
	required uint64 orderID = 2; //订单号
}

message Request
{
	required C2S c2s = 1;
}

message Response
{
	//以下3个字段每条协议都有，注释说明在InitConnect.proto中
	required int32 retType = 1 [default = -400];
	optional string retMsg = 2;
	optional int32 errCode = 3;
	
	optional S2C s2c = 4;
}
